### ISSUE TITLE
Error on windows named pipe connection

### ISSUE THREAD

**Issue by frozeman** [atom/electron#2703]
I try to establish a named pipe connection using

```
...
var socket  = new net.Socket();
socket.connect({path: '\\.\pipe\geth.ipc'});
socket.on('connect'...)
socket.on('error'...)
```

But the events never fire sometimes on windows 10

Any idea?

*Edit: not sure if this issue belongs here to to the node.js*

**Comment by frozeman** [comment]
Hm it seems to work when i use a setTimeout.

The full code is here:

This doesn't work, like the events are never fired (ps. i also moved the `socket.connect()` already below the events.)
```js
        const spawn = require('child_process').spawn;
        const getIpcPath = require('./modules/ipc/getIpcPath.js');
        const net = require('net');
        const socket = new net.Socket();
        var ipcPath = getIpcPath();
        var intervalId;
        var count = 0;

         // TRY to CONNECT
        socket.connect({path: ipcPath});

        socket.on('connect'...)
        socket.on('error'...)
```

This works:
```js
        const spawn = require('child_process').spawn;
        const getIpcPath = require('./modules/ipc/getIpcPath.js');
        const net = require('net');
        const socket = new net.Socket();
        var ipcPath = getIpcPath();
        var intervalId;
        var count = 0;

        setTimeout(function(){
            // TRY to CONNECT
            socket.connect({path: ipcPath});
        }, 1);

        socket.on('connect'...)
        socket.on('error'...)
```

Very odd

**Comment by zcbenz** [comment]
Which version of Electron are you using?

**Comment by zcbenz** [comment]
It looks very similar to a fixed bug(#1968), so I'm closing this until more information is provided.
